### PR TITLE
NAS-107152 / 12.1 / Retry gathering usage starts if we experience a failure

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/usage.py [hunk 1]
```diff
@@ -13,27 +13,35 @@
...
+                            headers={'Content-type': 'application/json'},
+                            proxy=os.environ.get('http_proxy'),
+                        )
+                except Exception as e:
+                    # We still want to schedule the next call
+                    self.middleware.logger.error(e)
+                    retries -= 1
+                    if retries:
+                        self.middleware.logger.debug('Retrying gathering stats after 5 minutes')
+                        await asyncio.sleep(300)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/usage.py:42]: 5 minutes is too low, we talked about an hour. Make it at least 30 minutes please.

File: src/middlewared/middlewared/plugins/usage.py [hunk 2]
```diff
@@ -13,27 +13,35 @@
...
+
+    FAILED_RETRIES = 3
+
     class Config:
         private = True
 
     async def start(self):
         if (
             await self.middleware.call('system.general.config')
         )['usage_collection']:
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/usage.py:25]: This should be within the loop too.

File: src/middlewared/middlewared/plugins/usage.py (hunk 1 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/usage.py:42]: No problem, we can have it at 30 minutes. It was not mentioned in the ticket at all so defaulted to 5 minutes but yes 30-60 min is indeed more reasonable.

File: src/middlewared/middlewared/plugins/usage.py [hunk 3]
```diff
@@ -13,27 +13,35 @@
...
-                        headers={"Content-type": "application/json"},
-                        proxy=os.environ.get("http_proxy"),
+                        data=await self.middleware.call('usage.gather'),
+                        headers={'Content-type': 'application/json'},
+                        proxy=os.environ.get('http_proxy'),
                     )
             except Exception as e:
                 # We still want to schedule the next call
-                self.logger.error(e)
+                self.middleware.logger.error(e)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/usage.py:38]: Why the change of logger?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/usage.py:38]: Consistency ? We normally use `middleware.logger.level`, i can update to the old one if you think that's best

Comment (william-gr) [reply src/middlewared/middlewared/plugins/usage.py:38]: Where do we use middleware.logger? Thats just wrong and we shouldnt be doing that for services.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/usage.py:38]: @sonicaj ping
